### ISSUE TITLE
Explicitly remove a worker from the pool

### ISSUE THREAD

**Issue by isaacabraham** [mbraceproject/MBrace.Azure#88]
Moved from MBrace.Core.

If you attach a local worker (for example), how can you remove it from the pool? If you just close the console window process, it still remains for a while as a worker in the pool. Work gets allocated to it, and jobs don't complete until the pool realises that the worker has died.

**Comment by krontogiannis** [comment]
Any workers (there is no distinction if workers runs on azure, or your machine or any machine) are considered active for a while, See https://github.com/mbraceproject/MBrace.Azure/issues/87.

AFAIK there is no way handle the kill signal for the local processes (when u click 'X' at the console window) and gracefully exit.

So we need sth like `runtime.KillLocalWorker(IWorkerRef)` that kills the local process and sets the worker Stopped?

**Comment by isaacabraham** [comment]
Just KillWorker and you pass in the worker ref. Doesn't need to be a local worker. This would also enable a "real" worker to send a kill message itself gracefully if / when the node goes down - particularly useful if you scale up / down workers on a cluster.

**Comment by krontogiannis** [comment]
It looks to me that the Brisk worker role doesn't call Service.Stop() on the WorkerRole's Stop method.
I just scaled a 10machine cluster down to 2:

```
runtime.ShowWorkers(showInactive = true)

 Workers                                                                                                                                                                                  

 Id               Status   % CPU / Cores  % Memory / Total(MB)  Network(ul/dl : KB/s)  Jobs   Hostname        Process Id  Heartbeat                   Initialization Time        
 --               ------   -------------  --------------------  ---------------------  ----   --------        ----------  ---------                   -------------------        
 WorkerRole_IN_2  Stopped     2.7 / 2        26.5 / 3583.0            2.3 / 2.4        0 / 2  RD000D3A20808C         976  15/06/2015 16:59:06 +00:00  15/06/2015 16:53:09 +00:00 
 WorkerRole_IN_7  Stopped     3.5 / 2        27.6 / 3583.0            2.9 / 2.5        0 / 2  RD000D3A208704        2320  15/06/2015 16:58:56 +00:00  15/06/2015 16:53:10 +00:00 
 WorkerRole_IN_9  Stopped     2.1 / 2        27.5 / 3583.0            2.6 / 2.7        0 / 2  RD000D3A208E27        2436  15/06/2015 16:59:04 +00:00  15/06/2015 16:53:22 +00:00 
 WorkerRole_IN_3  Stopped     1.7 / 2        29.5 / 3583.0            2.2 / 2.1        0 / 2  RD000D3A208988        3824  15/06/2015 16:59:03 +00:00  15/06/2015 16:53:28 +00:00 
 WorkerRole_IN_1  Running     2.4 / 2        26.7 / 3583.0            2.8 / 1.4        0 / 2  RD000D3A208B4E        3008  15/06/2015 16:59:29 +00:00  15/06/2015 16:53:32 +00:00 
 WorkerRole_IN_6  Stopped     1.7 / 2        27.4 / 3583.0            2.1 / 1.2        0 / 2  RD000D3A208AE5        2012  15/06/2015 16:58:25 +00:00  15/06/2015 16:53:40 +00:00 
 WorkerRole_IN_5  Stopped     3.0 / 2        26.7 / 3583.0            2.2 / 2.0        0 / 2  RD000D3A208664        1340  15/06/2015 16:57:57 +00:00  15/06/2015 16:53:50 +00:00 
 WorkerRole_IN_8  Stopped     3.5 / 2        26.7 / 3583.0            2.2 / 2.1        0 / 2  RD000D3A208760        3024  15/06/2015 16:59:03 +00:00  15/06/2015 16:54:01 +00:00 
 WorkerRole_IN_0  Running     1.3 / 2        26.7 / 3583.0            2.6 / 1.3        0 / 2  RD000D3A208958        2364  15/06/2015 16:59:29 +00:00  15/06/2015 16:54:08 +00:00 
 WorkerRole_IN_4  Stopped     2.6 / 2        27.4 / 3583.0            1.7 / 0.9        0 / 2  RD000D3A208F93        2248  15/06/2015 16:58:54 +00:00  15/06/2015 16:55:48 +00:00 
```

**Comment by isaacabraham** [comment]
Ok - will that do the equivalent of what I'm after on terms of full blown workers?
________________________________

**Comment by krontogiannis** [comment]
Yeah, `Service.Stop` updates the worker status to `Stopped`, and the workers and the worker is no longer returned in `GetWorkers` (or `Cloud.GetAvailableWorkers`, etc).

**Comment by dsyme** [comment]
It will be great to have this fixed - it does look like the same problem as https://github.com/mbraceproject/MBrace.Azure/issues/87

**Comment by isaacabraham** [comment]
I've updated the Brisk MBrace images this morning - any new clusters should now gracefully shut down. Can one of you try this out?

**Comment by isaacabraham** [comment]
@krontogiannis so if that works, yes it would be good to have the ability to explicitly detach workers from the pool, particularly for local workers (or have locally-attached workers auto-remove on shutdown).

**Comment by krontogiannis** [comment]
Since https://github.com/mbraceproject/MBrace.Azure/commit/2e02007fa5876bba0fc5486f1858766ca8b6d2c5 there is a `KillLocalWorker` method (two overloads, one for all local workers, and one that kills the given WorkerRef), also available in release v0.6.9, that kills local workers and sets their status as stopped (so that they don't show up in the `GetWorkers`).

**Comment by krontogiannis** [comment]
I just tried a Brisk cluster and any workers removed from the cluster immediately disappear form `GetWorkers` :)
